core/src/main/java/org/mule/runtime/core/internal/construct/AbstractPipeline.java

```diff
@@ -151,6 +158,8 @@ public AbstractPipeline(String name, MuleContext muleContext, MessageSource sour
     this.componentInitialStateManager = componentInitialStateManager;
     this.processors = unmodifiableList(processors);
     this.maxConcurrency = maxConcurrency != null ? maxConcurrency : DEFAULT_MAX_CONCURRENCY;
+    this.flowsSummaryStatistics = flowsSummaryStatistics;
+    this.triggerFlow = source != null || isApiKitFlow(getName());
```

Comment:
I think this should actually be `source != null || (source == null && isApiKitFlow(getName())`